wrappers/dlib/face/validate_face.cpp

```diff
@@ -0,0 +1,162 @@
+
+    float nose_depth;
+    if( ! find_depth_from( frame, depth_scale, face, 4, 4, &nose_depth ))
+        return false;
+
+    if( nose_depth >= eye_depth )
+    {
+        // This will still not work if you have a picture of a person that's tilted...
+        return false;
+    }
+    if( eye_depth - nose_depth > 0.07f )
+    {
+        // Too much distance... 7cm should be enough to cover all noses
+        return false;
+    }
+
+    return true;
+}
+
+
+bool validate_face_68(
+    rs2::depth_frame const & frame,
+    float const depth_scale,
+    dlib::full_object_detection const & face
+)
+{
+    // Collect all the depth information for the different facial parts
+
+    // For the ears, only one may be visible -- we take the closer one!
+    float left_ear_depth = 100, right_ear_depth = 100;
+    if( ! find_depth_from( frame, depth_scale, face, markup_68::RIGHT_EAR, markup_68::RIGHT_EAR+1, &right_ear_depth )
+    &&  ! find_depth_from( frame, depth_scale, face, markup_68::LEFT_EAR-1, markup_68::LEFT_EAR, &left_ear_depth ))
+        return false;
+    float ear_depth = std::min( right_ear_depth, left_ear_depth );
+
+    float chin_depth;
+    if( ! find_depth_from( frame, depth_scale, face, markup_68::CHIN_FROM, markup_68::CHIN_TO, &chin_depth ))
+        return false;
+
+    float nose_depth;
+    if( ! find_depth_from( frame, depth_scale, face, markup_68::NOSE_RIDGE_TO - 1, markup_68::NOSE_RIDGE_TO, &nose_depth ))
+        return false;
+
+    float right_eye_depth;
+    if( ! find_depth_from( frame, depth_scale, face, markup_68::RIGHT_EYE_FROM, markup_68::RIGHT_EYE_TO, &right_eye_depth ))
+        return false;
+    float left_eye_depth;
+    if( ! find_depth_from( frame, depth_scale, face, markup_68::LEFT_EYE_FROM, markup_68::LEFT_EYE_TO, &left_eye_depth ))
+        return false;
+    float eye_depth = std::min( left_eye_depth, right_eye_depth );
+
+    float mouth_depth;
+    if( ! find_depth_from( frame, depth_scale, face, markup_68::MOUTH_OUTER_FROM, markup_68::MOUTH_INNER_TO, &mouth_depth ))
+        return false;
+
+    // We just use simple heuristics to determine whether the depth information agrees with
+    // what's expected: that the nose tip, for example, should be closer to the camera than
+    // the eyes.
+    
+    // These heuristics are fairly basic but nonetheless serve to illustrate the point that
+    // depth data can effectively be used to distinguish between a person and a picture of a
+    // person...
+
+    if( nose_depth >= eye_depth )
+        return false;
+    if( eye_depth - nose_depth > 0.07f )
+        return false;
+    if( ear_depth <= eye_depth )
+        return false;
+    if( mouth_depth <= nose_depth )
+        return false;
+    if( mouth_depth > chin_depth )
+        return false;
+
+    float x = std::max( { nose_depth, eye_depth, ear_depth, mouth_depth, chin_depth } );
+    float n = std::min( { nose_depth, eye_depth, ear_depth, mouth_depth, chin_depth } );
+    if( x - n > 0.20f )
```

Comment:
Add an explanation for the heuristic

Reply:
// All the distances, collectively, should not span a range that makes no sense. I.e.,
    // if the face accounts for more than 20cm of depth, or less than 2cm, then something's
    // not kosher!